src/averageAges.js

```diff
@@ -38,6 +52,30 @@ function calculateMenAverageAge(people, century) {
  */
 function calculateWomenAverageAge(people, withChildren) {
   // write code here
+  const children = people.filter(child =>
+    child.mother !== null);
+  let findWomen = people.filter(listOfWomen =>
+    listOfWomen.sex === 'f');
+
+  if (arguments.length > 1) {
+    const listWithChildren = [];
+
+    for (let i = 0; i < children.length; i++) {
+      const trueMothers = people.find(list => list.name === children[i].mother);
+
+      if (!listWithChildren.includes(trueMothers)
+      && typeof trueMothers !== 'undefined') {
+        listWithChildren.push(trueMothers);
+      }
+    }
+    findWomen = listWithChildren;
+  }
+
+  const sumOfYears = findWomen.reduce((previousValue, years) =>
+    previousValue + (years.died - years.born), 0);
+  const getAvarage = Math.round((sumOfYears / findWomen.length) * 100) / 100;
```

Comment:
again, you do the same thing as in previous function, so make an external helper for that